internal/controller/controller.go

```diff
@@ -107,20 +107,22 @@ func (c *controller) Initialize(runtimeController ...runtimecontroller.Controlle
 
 	// Predicates are employed to filter out pod changes that are not necessary to reconcile.
 	if err := actualRuntimeController.Watch(
-		source.Kind(c.runtimeManager.GetCache(), &v1.Pod{}),
-		&handler.EnqueueRequestForObject{},
-		predicate.Funcs{
-			CreateFunc:  PredicateCreateFunc,
-			DeleteFunc:  PredicateDeleteFunc,
-			UpdateFunc:  PredicateUpdateFunc,
-			GenericFunc: PredicateGenericFunc,
-		},
+		source.Kind(
+			c.runtimeManager.GetCache(),
+			&v1.Pod{},
+			&handler.TypedEnqueueRequestForObject[*v1.Pod]{},
+			predicate.TypedFuncs[*v1.Pod]{
+				CreateFunc:  PredicateCreateFunc,
+				DeleteFunc:  PredicateDeleteFunc,
+				UpdateFunc:  PredicateUpdateFunc,
+				GenericFunc: PredicateGenericFunc,
+			},
+		),
 	); err != nil {
 		return common.WrapErrorf(err, "unable to watch pods")
 	}
 
-	csametrics.RegisterAllMetrics(metrics.Registry, Name)
-	defer csametrics.UnregisterAllMetrics(metrics.Registry)
+	csametrics.RegisterAllMetrics(metrics.Registry)
 	c.initialized = true
```

Comment:
Isn't this flag prone to synchronisation issues where [Once.Do](https://pkg.go.dev/sync#Once.Do) would be safer?

Reply:
In theory, yes. But at the moment this containing function is only ever invoked once by `main.go`. Will change to use `Once.Do` though!

Reply:
https://github.com/ExpediaGroup/container-startup-autoscaler/pull/14/commits/a3c8cc94790eea83db9585b22f7028518fd51eea